catroid/src/androidTest/java/org/catrobat/catroid/test/content/bricks/CloneCompositeBrickUpdateVariableTest.java

```diff
@@ -0,0 +1,177 @@
+import org.catrobat.catroid.content.StartScript;
+import org.catrobat.catroid.content.bricks.AddItemToUserListBrick;
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.ChangeVariableBrick;
+import org.catrobat.catroid.content.bricks.CompositeBrick;
+import org.catrobat.catroid.content.bricks.DeleteItemOfUserListBrick;
+import org.catrobat.catroid.content.bricks.ForeverBrick;
+import org.catrobat.catroid.content.bricks.IfLogicBeginBrick;
+import org.catrobat.catroid.content.bricks.RepeatBrick;
+import org.catrobat.catroid.content.bricks.SetVariableBrick;
+import org.catrobat.catroid.content.bricks.UserListBrick;
+import org.catrobat.catroid.content.bricks.UserVariableBrick;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.UserList;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.ui.recyclerview.controller.SpriteController;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+
+@RunWith(Parameterized.class)
+public class CloneCompositeBrickUpdateVariableTest {
+	private static final UserVariable USER_VARIABLE = new UserVariable("list");
+	private static final UserList USER_LIST = new UserList("list", Arrays.asList(new Object[]{"a", "b", "c"}));
+
+	@Parameterized.Parameters(name = "{0}")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{"SetVariableBrick", new SetVariableBrick(new Formula(2), USER_VARIABLE)},
+				{"ChangeVariableBrick", new ChangeVariableBrick(new Formula(2), USER_VARIABLE)},
+				{"AddItemToUserListBrick", new AddItemToUserListBrick(new Formula(2))},
+				{"DeleteItemOfUserListBrick", new DeleteItemOfUserListBrick(new Formula(2))}
+		});
+	}
+
+	@Parameterized.Parameter
+	public String name;
+
+	@Parameterized.Parameter(1)
+	public Brick brick;
+
+	private Sprite sprite;
+	private Script script;
+
+	@Before
+	public void setUp() {
+		Project project = new Project();
+		Scene scene = new Scene();
+		sprite = new Sprite("test_sprite");
+		scene.addSprite(sprite);
+		script = new StartScript();
+		sprite.addScript(script);
+		sprite.addUserVariable(USER_VARIABLE);
+		sprite.addUserList(USER_LIST);
+
+		project.addScene(scene);
+		ProjectManager.getInstance().setCurrentProject(project);
+
+		if (brick instanceof UserListBrick) {
+			((UserListBrick) brick).setUserList(USER_LIST);
+		}
+	}
+
+	@Test
+	public void testIf() {
+		IfLogicBeginBrick ifBrick = new IfLogicBeginBrick();
+		ifBrick.addBrickToIfBranch(brick);
+		script.addBrick(ifBrick);
+
+		Sprite cloneSprite = new SpriteController().copyForCloneBrick(sprite);
+		Brick cloneIfLogicBrick = cloneSprite.getScript(0).getBrick(0);
+		checkInNestedBrick((CompositeBrick) cloneIfLogicBrick);
```

Comment:
the function name could be more expressive. It's not clear, what you check. maybe checkCloneForNestedBrick or something better?